code/controllers/subsystem/garbage.dm

```diff
@@ -311,53 +341,58 @@ SUBSYSTEM_DEF(garbage)
 	else if(D.gc_destroyed == GC_CURRENTLY_BEING_QDELETED)
 		CRASH("[D.type] destroy proc was called multiple times, likely due to a qdel loop in the Destroy logic")
 
-#ifdef TESTING
-/client/var/running_find_references
+#ifdef REFERENCE_TRACKING
 
-/datum/verb/find_references()
+/client/proc/find_refs(datum/D in world)
 	set category = "Debug"
 	set name = "Find References"
-	set background = 1
-	set src in world
 
+	if(!check_rights(R_DEBUG))
+		return
+	D.find_references(FALSE)
+
+/datum/proc/find_references(skip_alert)
 	running_find_references = type
 	if(usr && usr.client)
 		if(usr.client.running_find_references)
-			testing("CANCELLED search for references to a [usr.client.running_find_references].")
+			log_gc("CANCELLED search for references to a [usr.client.running_find_references].")
 			usr.client.running_find_references = null
 			running_find_references = null
 			//restart the garbage collector
 			SSgarbage.can_fire = 1
 			SSgarbage.next_fire = world.time + world.tick_lag
 			return
 
-		if(tgui_alert(usr, "Running this will create a lot of lag until it finishes.  You can cancel it by running it again.  Would you like to begin the search?", "Find References", list("Yes", "No")) == "No")
-			running_find_references = null
-			return
+		if(!skip_alert)
+			if(tgui_alert("Running this will lock everything up for about 5 minutes.  Would you like to begin the search?", "Find References", "Yes", "No") == "No")
```

Comment:
потерял usr

Reply:
Так же, видимо, именно из-за этого тгуи алерта происходит слип в атом_ините.

![image](https://user-images.githubusercontent.com/26389417/126643604-31a7871b-f9ef-41ba-819a-c44a7ca05caf.png)

Reply:
```suggestion
			if(tgui_alert(usr, "Running this will lock everything up for about 5 minutes.  Would you like to begin the search?", "Find References", "Yes", "No") == "No")
```